phoenix/app/cad_products_phoenix/lib/cad_products_phoenix/elastik/product_index.ex

```diff
@@ -0,0 +1,49 @@
+defmodule CadProductsPhoenix.Elastik.ProductIndex do
+  import Tirexs.HTTP
+
+  def create_product(prod) do
+    product_json = product_json(prod)
+    post("/cad_products/products/#{product_json.id}", product_json)
+  end
+
+  def update_product(prod) do
+    product_json = product_json(prod)
+    put("/cad_products/products/#{product_json.id}", product_json)
+  end
+
+  def product_json(prod) do
+    product_json =
+        %{
+          id: prod.id,
+          sku: prod.sku,
+          name: prod.name,
+          price: prod.price,
+          qtd: prod.qtd,
+          description: prod.description,
+          barcode: prod.barcode,
+          last_update_at: DateTime.to_iso8601(DateTime.utc_now())
+        }
+    product_json
+  end
+
+  def delete_product(id) do
+    delete("/cad_products/products/#{id}")
+  end
+
+  def search_products(params) do
+    query = Enum.map_join(params, "*&", fn {k, v} -> "#{k}:#{v}" end)
```

Comment:
Faça um AND aqui. Se a pessoa passar um filtro como `name=teste&sku=123` e existir o name, mas não com esse sku, não é para retornar